tfgcv/planned_assets.go

```diff
@@ -59,3 +64,30 @@ func ReadPlannedAssets(path, project string) ([]google.Asset, error) {
 
 	return converter.Assets(), nil
 }
+
+var ErrParsingProviderProject = errors.New("unable to parse provider project")
+
+// parseProviderProject attempts to parse hardcoded "project" configuration
+// from the "google" provider block. It is lazy and fails if the job involves
+// interpolation.
+// TODO: Replicate/incorporate terraform interpolation (or is that a good idea?).
+func parseProviderProject(plan *terraform.Plan) (string, error) {
+	for _, cfg := range plan.Module.Config().ProviderConfigs {
+		if cfg.Name == "google" {
+			inf, ok := cfg.RawConfig.Raw["project"]
+			if !ok {
+				continue
+			}
+			prj := inf.(string)
+
+			// If the provider has a hardcoded project string, return it.
+			if !strings.Contains(prj, "${") {
+				return prj, nil
+			}
+
+			return "", ErrParsingProviderProject
+		}
+	}
+
+	return "", nil
```

Comment:
Why do we return nil for error instead of ErrParsingProviderProject?

Reply:
Provider `project` field is optional.

Reply:
... It is not an error in terraform if the `project` field is not specified. We are just error-ing out here if we are unable to parse it.

Reply:
Right, so I was curious about the difference in behavior. In both scenarios, we don't have a project id that we could use.
We return ErrParsingProviderProject when the google config is present and has a project id that is not hardcoded.
We return nil when the google config is not present at all. So if the --project flag is not set and there's no google config, we'll continue in line 50 to create a new Converter with an empty project string. I was a bit confused by that. Can we add a comment?

Reply:
Will do